php/class-coauthors-guest-authors.php

```diff
@@ -1503,4 +1506,93 @@ public function filter_author_feed_link( $feed_link, $feed ) {
 
 		return $link;
 	}
+
+	/**
+	 * Filter Personal Data Exporters to add Guest Author exporter
+	 *
+	 * @since 3.3.1
+	 */
+    public function filter_personal_data_exporter( $exporters ) {
+        $exporters['cap-guest-author'] = array(
+            'exporter_friendly_name' => __( 'Guest Author', 'co-authors-plus' ),
+            'callback'               => array( $this, 'personal_data_exporter' ),
+        );
+
+        return $exporters;
+    }
+
+	/**
+	 * Finds and exports personal data associated with an email address for guest authors
+	 *
+	 * @since 3.3.1
+	 *
+	 * @param string $email_address  The guest author email address.
+ 	 * @return array An array of personal data.
+	 */
+	public function personal_data_exporter( $email_address ) {
+		$email_address = trim( $email_address );
+
+		$data_to_export = array();
+
+		$author = $this->get_guest_author_by( 'user_email', $email_address );
+
+		if ( ! $author ) {
+			return array(
+				'data' => array(),
+				'done' => true,
+			);
+		}
+
+		$author_data = array(
+			'ID'              => __( 'Guest Author ID', 'co-authors-plus' ),
+			'user_login'      => __( 'Guest Author Login Name', 'co-authors-plus' ),
+			'display_name'    => __( 'Guest Author Display Name', 'co-authors-plus' ),
+			'user_email'      => __( 'Guest Author Email', 'co-authors-plus' ),
+			'first_name'      => __( 'Guest Author First Name', 'co-authors-plus' ),
+			'last_name'       => __( 'Guest Author Last Name', 'co-authors-plus' ),
+			'website'         => __( 'Guest Author Website', 'co-authors-plus' ),
+			'aim'             => __( 'Guest Author AIM', 'co-authors-plus' ),
+			'yahooim'         => __( 'Guest Author Yahoo IM', 'co-authors-plus' ),
+			'jabber'          => __( 'Guest Author Jabber / Google Talk', 'co-authors-plus' ),
+			'description'     => __( 'Guest Author Biographical Info', 'co-authors-plus' ),
+		);
+
+		$author_data_to_export = array();
+
+		foreach ( $author_data as $key => $name ) {
+			if ( empty( $author->$key ) ) continue;
+
+			$author_data_to_export[] = array(
+				'name'  => $name,
+				'value' => $author->$key,
+			);
+		}
+
+		/**
+		 * Filters extra data to allow plugins add data related to guest author
+		 *
+		 * @since 3.3.1
+		 *
+		 * @param array $extra_data A empty array to be populated with extra data
+		 * @param int $author->ID The guest author ID
+		 * @param string $email_address The guest author email address
+		 */
+		$extra_data = apply_filters( 'coauthors_guest_authors_exported_extra_data', [], $author->ID, $email_address );
```

Comment:
Instead of a separate array that we have to merge in, maybe we should just pass `$author_data_to_export` and other plugins can append to it directly?

Reply:
Also, for consistency, we should call this `coauthors_guest_author_personal_export_data`.

Reply:
Filter changed but I think we should keep the merge. This way we don't allow other plugins to remove our data, which I think is mandatory to keep because GDPR

Reply:
That makes sense @mariovalney. I think because of we should rename the filter to `coauthors_guest_author_personal_export_extra_data`.

Reply:
So... back to "coauthors_guest_author_personal_export_extra_data" ?

Reply:
Done